### PR TITLE
arrow: add

### PR DESCRIPTION
**Pull request by kou**
This is a new recipe for Apache Arrow: https://arrow.apache.org/

Apache Arrow is a columnar format for efficient in-memory data analytics.
Apache Arrow defines data format and provides libraries to process data in the format.

Apache Arrow supports many languages but this recipe supports only C++ and C (GLib). C++ library includes Python integration.

`check` is disabled for now because Apache Arrow doesn't support running tests with MinGW build yet.

### CODE REVIEW

File: mingw-w64-arrow/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,105 @@
+_realname=arrow
```
[the remark this answers is not in the document]
Comment (kou) [reply mingw-w64-arrow/PKGBUILD]: I've added "Mainatainer" header.

File: mingw-w64-arrow/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+             "${MINGW_PACKAGE_PREFIX}-gtk-doc"
+             "${MINGW_PACKAGE_PREFIX}-meson")
+options=("staticlibs" "strip" "!buildflags")
+source=("https://www.apache.org/dist/arrow/arrow-${pkgver}/apache-${_realname}-${pkgver}.tar.gz")
+sha256sums=('1838faa3775e082062ad832942ebc03aaf95386c0284288346ddae0632be855d')
+
+cmake_build_type=release
+meson_build_type=debugoptimized
+
+source_dir() {
```
[the remark this answers is not in the document]
Comment (kou) [reply mingw-w64-arrow/PKGBUILD]: I've changed to use variables instead of procedures.

**Comment by kou** [comment]
I'm trying to fix build error on 32bit environment.

**Comment by kou** [comment]
I've submitted a bug report to MinGW-w64 for g++ SEGV: https://sourceforge.net/p/mingw-w64/bugs/767/

**Comment by kou** [comment]
I've fixed all problems:

  * `g++` is crashed on 32bit build.
    * I've reported this to upstream (MinGW-w64): https://sourceforge.net/p/mingw-w64/bugs/767/
    * I've added workaround to upstream (Apache Arrow) and changed to use the patch in upstream: https://github.com/apache/arrow/commit/eb500b657895ed5f762aa7b819fcea2cb898358d.patch
  * We need to use `python2` or `python3` to find Python on MSYS2.
    * I've added a patch to fix this in upstream (Apache Arrow):  https://github.com/apache/arrow/commit/aaf70a41575f5eff590b4f73b8160ae896e6c24e.patch

@Alexpux Can you review again?

**Comment by dsisnero** [comment]
+1

**Pull request merged**